### DISCUSSION TITLE
Searching by field instances in Craft 5

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#14733] [General]
In Craft 5, we have an option to use multiple instances of specific field in field layout. However, this seems to have broken `search` element query method when it is used like this: `search('fieldInstance::some-value')`, as described in docs [here](https://craftcms.com/docs/5.x/system/searching.html#supported-syntaxes).

This is probably due to the fact the fact that in the `searchindex` db table, values of multiple instances of specific field are stored in single row for each element instead of using separate rows.

Are there any plans to allow searching by field instances? I know we can also just query like this `craft.entries.fieldInstance('some-value')`, but as far as i know, this method is not as performant as using `search`.

**Comment by brandonkelly** [comment]
We have a PR with per-instance search support (#13991), but I’m concerned about performance, so need to do some testing before merging it in.

**Comment by piotrpog** [comment]
@brandonkelly 
Any news about this?

**Reply by brandonkelly** [reply]
Nothing yet, sorry.
